app/scripts/app.js

```diff
@@ -485,6 +485,110 @@ angular
+      }, 60000);
+    };
+
+    // In case of multiple opened tabs the interval needs to be restarted everytime there is any activity in any tab,
+    // so all intervals are synced, and they start the counting at the same time.
+    var restartCheckInteractionInterval = function() {
+      // Handles cases of user activity during the opened modal, where user clicks inside the modal but not on the buttons.
+      if (modalInstance) {
+        modalInstance.dismiss("User activity");
+        modalInstance = null;
+      }
+      clearInterval(checkInteractionInterval);
+      startInteractionIntervalCheck();
+    };
+    // Updates lastInteraction date and restarts interval
+    var updateLastInteraction = function() {
+      restartCheckInteractionInterval();
+      localStorage.setItem(lastInteractionKey, new Date().toString());
+    };
+    // Updates Local Storage variable that indicates if all opened tabs should log out current user.
+    var logoutAllTabs = function(bool) {
+      localStorage.setItem(inactivityLogoutKey, bool.toString());
+    };
+    // Logout with passing reason.
+    var logout = function() {
+      var logoutURI = URI.expand("/logout{?cause*}", {
+        cause: 'timeout'
+      });
+      $location.url(logoutURI.toString());
+    };
+    // Show inactivity logout modal.
+    // Before the the modal is shown the handler for checking the changes in local storage is detached.
+    // After the the modal is closed or dismissed the handler will be attached again. This is because of 
+    // the fact that the handler is attached 
+    function showLogoutModal() {
+      if (isLocalLogoutModalShown) {
+        return false;
+      }
+      isLocalLogoutModalShown = true;
+      modalInstance = $uibModal.open({
+        animation: true,
+        templateUrl: 'views/modals/logout.html',
+        controller: 'LogoutModalController',
+      });
+      modalInstance.result.then(function(event) {
+        if (event === 'logout') {
+          logoutAllTabs(true);
+          logout();
+        } else if (event === 'cancel') {
+          // In case of multiple tabs, when the logoutModal would be canceled, the timer has to be restarted.
+          restartCheckInteractionInterval();
+          isLocalLogoutModalShown = false;
+        }
+      }, function () {
+        updateLastInteraction();
+        restartCheckInteractionInterval();
+        isLocalLogoutModalShown = false;
+      });
+    }
+
+    // Need to check for changes in last interaction so the interval can be restarted. 
+    $(window).on('storage', function (event) {
+      if (event.originalEvent.key === lastInteractionKey) {
+        restartCheckInteractionInterval();
+      } else if (event.originalEvent.key === inactivityLogoutKey) {
+        logout();
+      }
+    });
+
+    // Reset 'origin-web-console-inactivity-logout' key on login.
+    AuthService.onUserChanged(logoutAllTabs(false));
+    updateLastInteraction();
+    startInteractionIntervalCheck();
```

Comment:
same question as above, isnt the storage event going to fire and start the listener?